client/collection.go

```diff
@@ -78,21 +78,6 @@ type Collection interface {
 	// Will return true if a matching document exists, otherwise will return false.
 	Exists(ctx context.Context, docID DocID) (bool, error)
 
-	// UpdateWith updates a target document using the given updater type.
-	//
-	// Target can be a Filter statement, a single DocID, a single document,
-	// an array of DocIDs, or an array of documents.
-	// It is recommended to use the respective typed versions of Update
-	// (e.g. UpdateWithFilter or UpdateWithDocID) over this function if you can.
-	//
-	// Returns an ErrInvalidUpdateTarget error if the target type is not supported.
-	// Returns an ErrInvalidUpdater error if the updater type is not supported.
-	UpdateWith(
-		ctx context.Context,
-		target any,
-		updater string,
-	) (*UpdateResult, error)
-
 	// UpdateWithFilter updates using a filter to target documents for update.
```

Comment:
suggestion: Just `Update` and `Delete` would be better names IMO, especially given that the filtering is optional - they might be calling `UpdateWithFilter` with no filter.